### DISCUSSION TITLE
Is there a way to log a connection to Z-WAVE-SERVER ?

### DISCUSSION THREAD

**Discussion by goldriver2** [zwave-js/zwave-js-ui#3596] [Q&A]
Hi
I am converting some old Z-Net devices into Zwave-JS-UI embeded devices and I need a way to know if HomeAssistant is connected to the Z-WAVE-SERVER so I can change the status of the LED light on the Z-Net device depending on this connection.
I actually have two state, red ==> Controler not ready, yellow ==> Driver ready and would like to have a green ligth when HA is connected to the Z-WAVE-SERVER.

Actually in the log file, the only message I can see is when HA disconnects from ZWave-JS-UI:
_2024-02-10 20:57:50.685 INFO Z-WAVE-SERVER: Client disconnected_

Unless there is another way I can identify localy on the Raspberry running the ZWave_JS-UI if there is a connection between HA and Zwave-JS-UI that I am niot aware of

Thanks for the help on this

**Comment by robertsLando** [comment]
I dunno if zwave-js-server exposes this someway

CC @raman325

**Comment by raman325** [comment]
We emit a `listening` event that you could key off of to determine that the driver is ready. We don't have a shutdown/close event though and we don't have an event when a client connects/disconnect.

If you are concerned about the connection between `Home Assistant` and `zwave-js-server`, you can use a combination of three automations
1. Trigger on state change for the controller status entity from unavailable to not unavailable (means its connected)
2. Trigger on state change for the controller status entity from not unavailable to unavailable (means its disconnected)
3. Trigger on home assistant started - check the state of the controller status entity and act accordingly

**Comment by robertsLando** [comment]
> We emit a listening event that you could key off of to determine that the driver is ready. We don't have a shutdown/close event though and we don't have an event when a client connects/disconnect.

Humm ok but I would miss the state when the connection is lost too, dunno if the above suggestions could work for the user BTW. I think on HA side there should be something that show the integration is connected to the websocket server or not?

**Reply by goldriver2** [reply]
On HA side I can look at the zwave board device and find out the status BUT this is quite a detour way to just turn a LED on on the Raspberry when connection is established with Zwave-JS-UI that runs on another device...

For now, the ONLY way I found is to have ZWVA-JS-UI plugin configured with MQTT on the remote raspberry, then on HA (that sits on another machine) in Node-Red I created a node updating a topic with state of the device and finaly I build a script in the remote ZWAVE-JS-UI raspberry looping and watching both topics and reacting upon the changes...

Quite a detour only to have the status on the connection between HA and Zwave-JS-UI compared to a ZWave-JS-UI that would track client connect/disconnect and log it...

Could this be transfer as a request ?

**Reply by robertsLando** [reply]
I forward the question to @raman325 as he is the maintainer of the websocket server

**Comment by raman325** [comment]
> I think on HA side there should be something that show the integration is connected to the websocket server or not?

You can tell when the integration is connected to the websocket server because all of the entities that get created from the devices become available (they will be unavailable when the connection goes down). Hence my suggestion above.

I could add more events, but unless you plan to natively expose something in zjs-ui based on these events, I agree that it doesn't help in this case

**Reply by goldriver2** [reply]
I know on HA side I have ways to know if it is connected, my problem is that I am running zwave-js-ui on a separate device and I would have hope to have the connect/disconnect log in the zwave-js-ui zwave log file or anywhere locally where zwave-js-ui runs

**Comment by robertsLando** [comment]
> I could add more events, but unless you plan to natively expose something in zjs-ui based on these events, I agree that it doesn't help in this case

I think exposing events would not be so bad, I could at least log the changes to console and then think about showing the state on ui

**Comment by goldriver2** [comment]
That would be perfect
